docker/tls-setup/gencerts.sh

```diff
@@ -0,0 +1,221 @@
+#!/usr/bin/env bash
```

Comment:
would this be better served as being a small go program (more testable / no deps on various openssl tooling etc)

Reply:
We can merge this PR as it stands - but I'd like to see a new issue to handle certificate generation a little more robustly in the future.

Reply:
Uhm, @benbromhead would you mind elaborating a bit here? We're creating the keystores _with_ the keys using the java's `keytool`. The only reason we use openssl _specifically_ is to pull the key out of the keystore to configure `cqlsh` to connect to the local node.

If we were to rewrite things here in golang....do you mean rewrite the whole chain, from generating certs, to CSRs, to keystores? Not sure golang has implementation to create keystore files yet.

As I see it, rewriting it in golang doesn't really provide us with any benefit nor makes it more "robust",  so I'd love to understand more.

Reply:
iirc java also supports PKCS12 (https://docs.oracle.com/javase/7/docs/technotes/guides/security/StandardNames.html#KeyStore) as a keystore format which is an open format most crypto libs should be able to write out.

some quick thoughts off the top of my head on why writing robust bash is hard:
- fail first is off by default... by default scripts will keep executing if a command hits an error, you need to do `set -e` (not done in this script as well) to turn off this behavior 
- dependencies are scoped at the OS level, rather than the libraries compiled into the binary, so if something gets upgraded (e.g. one of the tools / executables the script calls) it could break
- variables are global to the user env, rather than the program
- lacking good test frameworks, hard to test etc